From b0fec92ea8a480b5bb595f0aedb48b488b135176 Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Thu, 1 Dec 2016 09:51:35 +0800
Subject: [PATCH 245/291] AOSCOS: Input: i8042 - Make i8042_bypass_aux_irq_test
 as a module parameter

The original i8042_bypass_aux_irq_test is a variable which cannot be
changed without kernel rebuild. Some buggy platform fails to detect the
existance of AUX if i8042_bypass_aux_irq_test isn't set. So make this
variable as a module parameter and set the default value to true.

Signed-off-by: Huacai Chen <chenhc@lemote.com>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/input/serio/i8042.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/input/serio/i8042.c b/drivers/input/serio/i8042.c
index 199e9f83a165..39175a94a7f3 100644
--- a/drivers/input/serio/i8042.c
+++ b/drivers/input/serio/i8042.c
@@ -131,7 +131,9 @@ MODULE_PARM_DESC(unmask_kbd_data, "Unconditional enable (may reveal sensitive da
 #endif
 
 static bool i8042_present;
-static bool i8042_bypass_aux_irq_test;
+static bool i8042_bypass_aux_irq_test = true;
+module_param_named(bypass_aux_test, i8042_bypass_aux_irq_test, bool, 0600);
+MODULE_PARM_DESC(debug, "Turn i8042 aux irq test off");
 static char i8042_kbd_firmware_id[128];
 static char i8042_aux_firmware_id[128];
 static struct fwnode_handle *i8042_kbd_fwnode;
-- 
2.48.1

